### DISCUSSION TITLE
Support foundationDB

### DISCUSSION THREAD

**Discussion by markus-walther** [oven-sh/bun#10123] [Ideas]
### What is the problem this feature would solve?

https://www.foundationdb.org/ is the most trustable open-source distributed database out there, e.g. used for Apple iCloud storage.

The problem this would solve is what could be a true differentiator for bun.

Bun could truely differentiate itself from the rest of the pack by integrating a fast API to use foundationDB, and offer foundationDB on the edge for when it comes to making money. The criticism of what the others do is laid out in https://kerkour.com/distributed-sqlite.

### What is the feature you are proposing to solve the problem?

Have a fast, DX-friendly API to foundationDB, so bun can be used out-of-the-box with foundationDB.

Equivalently, add foundationDB support via an easy-to-install plugin.

### What alternatives have you considered?

There are no alternatives that match foundationDB's triple offering of being distributed, best-tested-database and strictly serializable transactions.

**Comment by constraintAutomaton** [comment]
I don't understand it seems like someone should create a Javascript interface for the foundationDB API using ( or running it with) bun and not a question of adding new functionality for Bun?

**Comment by markus-walther** [comment]
@constraintAutomaton That is incorrect. Bun already tightly integrates an optimized interface to an embedded database, sqLite, as a precedent. This issue asks to add a _particular_ more powerful database in a similar fashion, for the reasons outlined. The way this is done is left open, I mentioned the possibility of a plugin.

**Reply by ghiscoding** [reply]
potentially related to PR #6531 for a new `bun:sql` (starting with Postgres)

**Reply by cloudspeech** [reply]
FoundationDB, as the name suggests, is a foundation for databases (more correctly, database layers). One such layer could be a SQL layer (they have been written). So, not directly related. Using foundationDB directly maps most easily to a key-value data base, i.e. NoSQL data base in older terminology.

**Comment by constraintAutomaton** [comment]
> @constraintAutomaton That is incorrect. Bun already tightly integrates an optimized interface to an embedded database, sqLite, as a precedent. This issue asks to add a _particular_ more powerful database in a similar fashion, for the reasons outlined. The way this is done is left open, I mentioned the possibility of a plugin.

I should have looked more carefully my bad.

**Comment by cloudspeech** [comment]
https://deno.com/blog/building-deno-kv#foundationdb-scalable-distributed-flexible-and-production-ready

and

https://www.bismuthos.com/  together with their comment on Hacker News https://news.ycombinator.com/item?id=40322094

suggest my feature request is right on the mark...

**Comment by Jarred-Sumner** [comment]
We are unlikely to add official support for FoundationDB, but there are probably npm packages which expose it
